service/src/main/java/uk/gov/hmcts/reform/fpl/service/LocalAuthorityService.java

```diff
@@ -31,13 +32,14 @@
     private final LocalAuthorityIdLookupConfiguration idsConfig;
     private final LocalAuthorityCodeLookupConfiguration codesConfig;
     private final LocalAuthorityNameLookupConfiguration namesConfig;
+    private final OrganisationService organisationService;
 
     public Optional<String> getLocalAuthorityCode() {
         UserInfo userInfo = idam.getUserInfo(requestData.authorisation());
         String email = userInfo.getSub();
         String domain = extractEmailDomain(email);
 
-        return codesConfig.getLocalAuthorityCode(domain);
+        return codesConfig.getLocalAuthorityCode(domain).or(this::getLocalAuthorityFromOrganisation);
```

Comment:
Would be it be better to rely on PRD as the source of truth first then fall back to our local mapping? Or is this way better due to less network requests or was what business wanted?

Reply:
PRD for now has been decided by business to be only fallback solution played in situation when we cannot do correct mapping. Main reason is that relying on PRD reverse order of onboarding process. Currently we can onboard LA that did not register in MO. If we treat PRD as main source of data then org should be first onboarded in PRD before we onboard on our side